JS/Script.js

```diff
@@ -1,73 +1,100 @@
+    this.addBook = document.querySelector('#add-book');
+    this.library = {};
+  }
 
-function storeEntryData(titleElement, authorElement) {
-  let entryObject = {};
-  entryObject.title = titleElement.value;
-  entryObject.author = authorElement.value;
-  library.push(entryObject);
-}
+  storeEntryData = (titleElement, authorElement) => {
+    const entryObject = {};
+    entryObject.title = titleElement.value;
+    entryObject.author = authorElement.value;
+    this.library.push(entryObject);
+  }
 
-function appendBookHTML(title = false, author = false) {  
-  if (title === false || author === false) { return }
-  let newBook = document.createElement('div');
-  let newTitle = document.createElement('div');
-  newTitle.innerHTML = title;
-  newBook.appendChild(newTitle);
-  let newAouthor = document.createElement('div');
-  newAouthor.innerHTML = author;
-  newBook.appendChild(newAouthor);
-  let removeButton = document.createElement('div');
-  removeButton.innerHTML = '<button class="remove">Remove</button><br><br>';
-  newBook.appendChild(removeButton);
-  bookCollection.appendChild(newBook);
-} 
+  appendBookHTML = (title = false, author = false) => {
+    if (title === false || author === false) {
+      return true;
+    }
+    const newBook = document.createElement('div');
+    newBook.setAttribute('class', 'book-list');
+    const newTitle = document.createElement('div');
+    newTitle.innerHTML = title.concat('-by-');
+    newBook.appendChild(newTitle);
+    const newAouthor = document.createElement('div');
+    newAouthor.innerHTML = author;
+    newBook.appendChild(newAouthor);
+    const removeButton = document.createElement('div');
+    removeButton.innerHTML = '<button class="remove">Remove</button><br><br>';
+    newBook.appendChild(removeButton);
+    this.bookCollection.appendChild(newBook);
+    return false;
+  }
 
-function loadBooks() {
-  for (let i = 0; i < library.length; i++) {    
-    appendBookHTML(library[i].title, library[i].author)
+  loadBooks = () => {
+    for (let i = 0; i < this.library.length; i += 1) {
+      this.appendBookHTML(this.library[i].title, this.library[i].author);
+    }
   }
-}
 
-loadBooks();
+  updateNewBook = () => {
+    // eslint-disable-next-line max-len
+    this.appendBookHTML(this.library[this.library.length - 1].title, this.library[this.library.length - 1].author);
+    localStorage.setItem('Awesome Book Collection', JSON.stringify(this.library));
+  }
 
-function updateNewBook() {
-  appendBookHTML(library[library.length-1].title, library[library.length-1].author);
-  localStorage.setItem('Awesome Book Collection', JSON.stringify(library));
+  removeBook = (element) => {
+    const parent = element.parentElement.parentElement;
+    const parrentIndex = Array.from(this.bookCollection.children).indexOf(parent);
+    parent.remove();
+    this.library.splice(parrentIndex, 1);
+    localStorage.setItem('Awesome Book Collection', JSON.stringify(this.library));
+  }
```

Comment:
- [OPTIONAL] I would advise you to use the `filter` method instead `splice` method. The `splice` method mutates arrays. Mutations may lead to unexpected and hard-to-debug issues, where data becomes incorrect somewhere, and you have no idea where it happens.